### ISSUE TITLE
Angular2 tree not expanding by default if the entire tree is changed

### ISSUE THREAD

**Issue by MjRockss** [500tech/angular-tree-component#520]
I am working on angular2 tree component where initially I am displaying the tree as onInit

```
var nodes =
[
    {
        name: 'a',
        isExpanded: true,
        children: [
            {
                name: 'b',
                isExpanded: true,
                children: [
                    { name: 'c', }
                ]
            },
        ]
    },
    {
        name: 'b',
        isExpanded: true,
        children: [
            {
                name: 'a',
                children: [
                    { name: 'c' },
                ]
            }
        ]
    }
];
```
this works absolutely fine and the tree is displayed in expanded, on a button click i am changing the entire tree and the new tree looks something like this

```
 changeTree{
  nodes =
        [
            {
                name: 'new Node a',
                isExpanded: true,
                children: [
                    {
                        name: ' new Node b',
                        isExpanded: true,
                        children: [
                            { name: 'new Node c', }
                        ]
                    },
                ]
            },
            {
                name: 'new Node b',
                isExpanded: true,
                children: [
                    {
                        name: 'new Node a',
                        children: [
                            { name: ' new Node c' },
                        ]
                    }
                ]
            }
        ];

this.tree.treeModel.update(); 
this.tree.treeModel.expandAll();
}
```
I tried using **this.tree.treeModel.update();** and **this.tree.treeModel.expandAll();** but still the tree is not getting expanded when a new tree is loaded.

Any help would be appreciated.

**Comment by erickyi2006** [comment]
i found this too. i suspect that the actual treeModel is also changed once u have updated your nodes.

have you tried this? in your onButtonClicked event, pass the tree as part of your parameter e.g.
`<button (click)="onButtonClicked(tree, $event)" >blah blah </button>`
Then use the passed in tree instead of this.tree 

assumption: you have declared #tree in the html.

**Comment by samartioli** [comment]
Looks like this is the issue... 
https://github.com/500tech/angular-tree-component/blob/master/lib/models/tree.model.ts#L212-L215
```
if (this.firstUpdate) {
      if (this.roots) {
        this.firstUpdate = false;
        this._calculateExpandedNodes();
      }
```

only runs `_calculateExpandedNodes()` first time through... 

as a hack you can do:

```
this.tree.treeModel.update();
this.tree.treeModel._calculateExpandedNodes();
```

but typescript will complain about calling a private method

**Comment by adamkleingit** [comment]
Thanks.
Merged and published in 8.1.0
